pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -244,11 +247,119 @@ func (f *Fit) EventsToRegister() []framework.ClusterEventWithHint {
 		podActionType |= framework.Update
 	}
 	return []framework.ClusterEventWithHint{
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: podActionType}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: podActionType}, QueueingHintFn: f.isSchedulableAfterPodChange},
+		// TODO: Ideally, it's supposed to add a QueueingHintFn for verifying whether the Pod becomes schedulable after receiving
+		// events of NodeCreated and NodeUpdated. But, we may miss NodeCreated event due to preCheck. It may cause the Pod to be
+		// stuck in the scheduling queue forever.
+		// See: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/119177/pull-kubernetes-integration/1726440712578273280
 		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}},
+		// TODO: The following line is expected to be uncommented once the preCheck on NodeEvent is removed.
+		// {Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}, QueueingHintFn: f.isSchedulableAfterNodeChange},
 	}
 }
 
+// isSchedulableAfterPodChange is invoked whenever a pod deleted or updated. It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (f *Fit) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalPod, modifiedPod, err := schedutil.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	if modifiedPod == nil {
+		logger.V(4).Info("an other pod was deleted", "pod", klog.KObj(pod), "otherPod", klog.KObj(originalPod))
+		return framework.Queue, nil
```

Comment:
Do we need it? (In other words, does this plugin need to subscribe the deletion event of Pods?)
When a Pod is deleted, the Node's allocatable resource should be increased right after that (= NodeUpdated event).

Reply:
@sanposhiho 

when a pod is deleted, it won't be triggered a node update event because the allocatable is not changed.

node allocatable resource (cpu, memory, etc.) is evaluated at kubelet startup phase. 
plugin allocatable resource is evaluated periodically from a goroutine.

so we also need it.

Reply:
Ah correct. I meant to say `NodeInfo.Requested` though, but `NodeInfo.Requested` seems to be calculated in the scheduler. (I thought `NodeInfo.Requested` was coming from some status field of Node.)

OK understood, let's keep it.